src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        int figureIndex = random.nextInt(0, 5);
+
+        switch (figureIndex) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(0, 10));
+            case 1:
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(0, 10));
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(), random.nextInt(0, 10),
+                        random.nextInt(0, 10));
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(0, 10),
+                        random.nextInt(0, 10));
+            case 4:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(), random.nextInt(0,10),
+                        random.nextInt(0, 10));
+            default:
+                System.out.println(" ");
```

Comment:
throw new IllegalStateException for example, or just replace case 4 with default

Reply:
not fixed